src/cleanvision/videolab.py

```diff
@@ -0,0 +1,260 @@
+"""Videolab is an extension of Imagelab for finding issues in a video dataset."""
+from pathlib import Path
+from typing import Any, Dict, Generator, List, Optional
+
+import av
+import pandas as pd
+from PIL.Image import Image
+
+from cleanvision.imagelab import Imagelab
+from cleanvision.utils.utils import get_is_issue_colname
+
+VIDEO_FILE_EXTENSIONS = ["*.mp4", "*.avi", "*.mkv", "*.mov", "*.webm"]
+
+
+class FrameSampler:
+    """Simplest frame sampling strategy."""
+
+    def __init__(self, k: int) -> None:
+        """Store frame sample interval k."""
+        self.k = k
+
+    def _create_frame_sample_sub_dir(self, video_file: Path, output_dir: Path) -> Path:
+        """Create a unique sub direcotry for storing frame samples from a video file."""
+        # create new sub directory from video file name
+        sub_dir = output_dir / video_file.name
+        sub_dir.mkdir(parents=True)
+
+        # return path to new sub dir
+        return sub_dir
+
+    def sample(self, video_file: Path, output_dir: Path) -> None:
+        """Loop through frames and store every k-th frame."""
+        # create frame samples sub directory
+        sample_sub_dir = self._create_frame_sample_sub_dir(video_file, output_dir)
+
+        # open video file for streaming
```

Comment:
```suggestion
        # open video file for streaming
        try:
            import av
        except ImportError as error:
            raise ImportError(
                "Cannot import package `av`. "
                "Please install it via `pip install av` and then try again."
             ) from error
        import av
```

Reply:
Can lazy import the optional dependency `av` only where it's required

Reply:
Before I commit these changes I must clarify: should the *import* be at the top of the source file with the other imports? Like this:
```python
"""Videolab is an extension of Imagelab for finding issues in a video dataset."""
from pathlib import Path
from typing import Any, Dict, Generator, List, Optional

 try:
    import av
except ImportError as error:
    raise ImportError(
        "Cannot import package `av`. "
        "Please install it via `pip install av` and then try again."
     ) from error
import av

import pandas as pd
from PIL.Image import Image

from cleanvision.imagelab import Imagelab
from cleanvision.utils.utils import get_is_issue_colname
```

What are the benefits of importing it directly in the `sample` method? This method will be called for each video. Will that not be a performance issue? Just unfamiliar with this strategy :sweat_smile: ...

Reply:
You can move this line of code wherever is the first time a user would do something related to videolab. I.e. when they construct the object.

The reason I don't think this line of code can be at the top of the file is because the file will be imported with: import cleanvision

And thus this exception will be raised, even for users who didn't want to use Videolab.

So you can really put this code anywhere that makes it possible for users to run Imagelab without having av installed (which you can easily test yourself), while ensuring the exception gets raised as soon as such users try to run Videolab.

Reply:
Excellent, I see. I will do exactly that. Thank you for clarifying.